### DISCUSSION TITLE
page does not refresh after deleting card

### DISCUSSION THREAD

**Discussion by granitn** [neuefische/questions#599] [Web] [answered]
Hello,

I managed to get the button on each card to work as far a to delete the entry from my database but the deletion is only displayed after manually reloading the page.
After about a hour of googling and even trying chatGPT,
i know I'm missing something small but cannot find out what.

I have committed and pushed my current [status](https://github.com/granitn/capstone-project/pull/7) . The branch is called deletion-feature

**Comment by mbosselmann** [comment, marked as answer]
Hi Granit,
@actyralf and I have found the "something small" that you can use to get what you want. You can work with `mutate` from useSWR to trigger a refetch of your data. [See the SWR docs.](https://swr.vercel.app/docs/mutation#revalidation)
Instead of using `router.push` call `mutate()`:

```jsx
export default function MachineCard({ machine }) {
  const { mutate } = useSWR("/api/machines");

  async function handleDelete() {
    const response = await fetch(`/api/machines/${machine._id}`, {
      method: "DELETE",
    });

    if (response.ok) {
      await response.json();
      mutate();
    } else {
      console.error(`Error: ${response.status}`);
    }
  }

  return (
...
  );
}
```

I hope this does help! :)
